### PR TITLE
feat(storage-agreements): handles payout confirmations

### LINKED ISSUE #610: Handle payoutFunds confirmations
While awaiting confirmations after `payoutFunds` action, we should show the spinner on `available funds` and `withdraw`

### PR DESCRIPTION
**Pull request by farinalvaro**
- Fixes payoutFunds method by ensuring proper length for encoded hashes
- Adds confirmations functionality to payoutFunds action
Closes #610

### CODE REVIEW

File: src/contracts/storage/utils.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function prefixArray(
 
 export function encodeHash(hash: string): string[] {
   if (hash.length <= 32) {
-    return [asciiToHex(hash)]
+    return [asciiToHex(hash).padEnd(66, '0')]
```
Comment (farinalvaro) [inline src/contracts/storage/utils.ts:48]: Just for clarification, the 66 is because of the `0x`

**Review by julianmrodri — APPROVED** [review]
LGTM

**Review by jurajpiar — APPROVED** [review]

**Comment by farinalvaro** [comment]
Converting to draft so I can add the new useConfirmations hook added on https://github.com/rsksmart/rif-marketplace-ui/pull/625 before merging

**Review by julianmrodri — APPROVED** [review]
LGTM

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**